rsts/concepts/lifecycle_workflow_with_plugin.rst

```diff
@@ -0,0 +1,252 @@
+.. _lifecycle-workflow:
+
+#################################################################
+Understand the Lifecycle of a Flyte Workflow
+#################################################################
+
+After reading this entry you should be able to understand how Flyte's Plugin machinery
+work under the hood and how information flows from the different
+components in Flyte.
+
+Under the hood Flyte relies on a primitive called “Plugins”. Every task
+that you run on Flyte is somehow powered by a plugin. Some of those plugins are native and guaranteed by Flyte system. These native plugins are for example powering your python functions (i.e: Tasks) by running them inside a k8s pod. There are Three native plugins : ``Container``, ``K8sPod`` and ``Sql``.
+
+Addionally there are also plugins which are actual extensions, they create additional infrastructure, talk to other SaaS on your behalf. Examples of those are the Spark or AWS Athena plugins.
+
+A plugin requires code to live in multiple locations
+
+1. Some plugin logic has to live in SDK so that users can create tasks
+   that make use of a plugin. This logic usually goes into a specific
+   SDK i.e: flytekit’s python: https://github.com/flyteorg/flyteplugins
+
+2. Another big chunk of plugins logic lives in
+   `Flyteplugins <https://github.com/flyteorg/flyteplugins>`__. This
+   repository is a library that gets loaded into
+   `FlytePropeller <https://github.com/flyteorg/flytepropeller>`__.
+   Flytepropeller is a Kubernetes operator, when Flytepropeller starts
```

Comment:
This is a library that is loaded into `FlytePropeller <https://github.com/flyteorg/flytepropeller>`__.